pkg/certificate/service.go

```diff
@@ -192,10 +192,6 @@ func New(ctx context.Context, idCfg *config.IdentityConfig) (daemon.Daemon, erro
 			if err != nil {
 				log.Errorf("Failed to retrieve x509 certificate from identity provider: %s", err.Error())
 			}
-			err = idCfg.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM)
-			if err != nil {
-				log.Errorf("Failed to reload x509 certificate from identity provider: %s", err.Error())
-			}
 		} else if idCfg.KeyFile != "" && idCfg.CertFile != "" {
 			log.Debug("Attempting to load x509 certificate from cert reloader...")
 			localFileKeyPEM, localFileCertPEM, err := idCfg.Reloader.GetLatestKeyAndCert()
```

Comment:
This part read from the reloader.
After this refactoring, it will save back to the reloader. I think will need be more careful on what may be affected and extend the scope of refactoring.

Reply:
I think we can do `idCfg.Reloader.UpdateCertificate` only when it is non-local-file reading, then.

Reply:
Maybe it is better to merge `pkg/util/tls-reload.go` ONLY to fix the bug first.

And leave the refactoring of `pkg/certificate/service.go` later on?
We also have another refactoring related to the reloader. https://github.com/AthenZ/k8s-athenz-sia/issues/109

Reply:
> And leave the refactoring of pkg/certificate/service.go later on?

This is not a refactor. we actually found two separate bugs.

Reply:
> Maybe it is better to merge pkg/util/tls-reload.go ONLY to fix the bug first.

You are right. The following pr will fix it.
https://github.com/AthenZ/k8s-athenz-sia/pull/118

Reply:
> And leave the refactoring of pkg/certificate/service.go later on?
Again this is not a refactor but a bug fix pr. so it is a must.